crates/astria-sequencer/src/sequencer.rs

```diff
@@ -210,6 +188,83 @@ fn start_grpc_server(
     )
 }
 
+fn start_abci_server(
+    storage: &cnidarium::Storage,
+    app: App,
+    mempool_service: service::Mempool,
+    listen_url: &str,
+    server_exit_tx: oneshot::Sender<()>,
+) -> Result<JoinHandle<()>, Report> {
+    // Setup services required for the ABCI server
+    let consensus_service = tower::ServiceBuilder::new()
+        .layer(request_span::layer(|req: &ConsensusRequest| {
+            req.create_span()
+        }))
+        .service(tower_actor::Actor::new(10, |queue: _| {
+            let storage = storage.clone();
+            async move { service::Consensus::new(storage, app, queue).run().await }
+        }));
+    let info_service =
+        service::Info::new(storage.clone()).wrap_err("failed initializing info service")?;
+    let snapshot_service = service::Snapshot;
+
+    // Builds the server but does not start listening.
+    let server = Server::builder()
+        .consensus(consensus_service)
+        .info(info_service)
+        .mempool(mempool_service)
+        .snapshot(snapshot_service)
+        .finish()
+        .ok_or_eyre("server builder didn't return server; are all fields set?")?;
+
+    // Validate and parse the listen_url received from the config.
+    let abci_url = Url::parse(listen_url).wrap_err("failed to parse listen_addr")?;
+    let validated_listen_addr = match abci_url.scheme() {
+        "unix" => match abci_url.to_file_path() {
+            Ok(_) => Ok(abci_url.path().to_string()),
+            Err(e) => Err(eyre!(
+                "failed parsing unix listen_addr file path, error: {:?}",
```

Comment:
I figure `eyre::WrapErr::wrap_err` was not doable because this is a boxed error? We have solved a related issue in `astria_eyre` by wrapping the boxed error. You can then construct a proper chain:

```rust
eyre!(e).wrap_err("failed rasing unix listen_addr file path")
```